src/components/chat/chat_channels.jsx

```diff
@@ -276,6 +270,10 @@ const ChatChannels = React.createClass({
       channels: ChatChannelsStore.getChannels(),
       subscribedChannels: ChatChannelsStore.getSubscribedChannels()
     });
+    // Fetch the initial list of messages for all subscribed rooms.
+    ChatChannelsStore.getSubscribedChannels().map((channel) => {
```

Comment:
Won't this potentially trigger multiple updates, especially because it runs whenever the channels change? I think we might be running into some pretty complicated logic surrounding how/what messages to fetch one. It seems to me that other chat clients get around this by only keeping the currently viewed chat room's messages in memory; I like that fetching all subscriptions will keep everything feeling snappy, but I'm not yet sure how best to manage the tradeoffs.